src/views/Rewards/Rewards.js

```diff
@@ -1,132 +1,199 @@
+  // });
+
+  const { data: rewardWeeks, error: failedFetchingRewards } = useSWR([getTracerServerUrl(chainId, "/rewards")], {
+    fetcher: (...args) => fetch(...args).then((res) => res.json()),
+  });
+
+  const userData = useMemo(
+    () =>
+      rewardWeeks?.reduce(
+        (totals, week) => {
+          const trader = week.traders.find((trader) => trader.user_address === account);
+          if (!trader) {
+            return totals;
+          }
+          return {
+            totalTradingVolume: totals.totalTradingVolume.add(trader.volume),
+            totalRewards: totals.totalRewards.add(trader.reward),
+            unclaimedRewards: totals.unclaimedRewards.add(trader?.claimed ? trader.amount : 0),
+          };
+        },
+        {
+          totalTradingVolume: ethers.BigNumber.from(0),
+          totalRewards: ethers.BigNumber.from(0),
+          unclaimedRewards: ethers.BigNumber.from(0),
+        }
+      )[(rewardWeeks, account)]
+  );
+
+  const weekData = useMemo(() => {
+    if (!rewardWeeks) {
+      return undefined;
+    }
+    if (!!rewardWeeks?.message) {
+      return undefined;
+    }
+    const selectedWeekActual = selectedWeek && parseFloat(selectedWeek) - 1;
+    const weekData = rewardWeeks?.find((week) => week.week === selectedWeekActual?.toString());
+    if (!weekData) {
+      return undefined;
+    }
+    weekData.traders.sort((a, b) => b.volume - a.volume); // Sort traders by highest to lowest in volume
+    return weekData;
+  }, [rewardWeeks, selectedWeek]);
+
+  const userWeekData = useMemo(() => {
+    if (!weekData) {
+      return undefined;
+    }
+    const traderData = weekData.traders?.find((trader) => trader.user_address === account);
+    const leaderboardPosition = weekData.traders?.findIndex((trader) => trader.user_address === account);
+    // trader's data found
+    if (traderData) {
+      traderData.position = leaderboardPosition;
+      return traderData;
+    } else {
+      // trader not found but data exists so user has no rewards
+      return {
+        volume: ethers.BigNumber.from(0),
+        reward: ethers.BigNumber.from(0),
+      };
+    }
+  }, [account, weekData]);
 
-  const [rewardsWeek, setRewardsWeek] = useState(REWARD_WEEKS[0].key);
+  const eth = getTokenInfo(infoTokens, ethers.constants.AddressZero);
+  const ethPrice = eth?.maxPrimaryPrice;
+
+  let rewardAmountEth = 0;
+  if (ethPrice && userWeekData) {
+    rewardAmountEth = ethPrice.mul(userWeekData.reward);
+  }
+
+  let unclaimedRewardsEth, totalRewardAmountEth;
+  if (ethPrice && userData) {
+    unclaimedRewardsEth = ethPrice.mul(userData.unclaimedRewards);
+    totalRewardAmountEth = ethPrice.mul(userData.totalRewards);
+  }
+
+  if (!rewardWeeks && selectedWeek !== undefined) {
+    setSelectedWeek(undefined);
+  } else if (selectedWeek === undefined && !!rewardWeeks) {
+    setSelectedWeek(parseFloat(rewardWeeks[rewardWeeks.length - 1].week) + 1);
+  }
```

Comment:
This might change in a bit since rewards will already be in ETH. Will use this to convert to USD but fine for now